### PR TITLE
Refactoring MOBILE var :dizzy:

### PR DESCRIPTION
**Pull request by kosssi**
We can have NODE_ENV for an application Cozy (browser, mobile (cordova), desktop (electron)):

1)
```
browser:
- development
- production

mobile:
- mobile
- mobile_production

desktop:
- desktop
- desktop_production
```

or

2)
```
browser:
- development
- production

mobile:
- mobile_development
- mobile_production

desktop:
- desktop_development
- desktop_production
```

or

3)
```
browser:
- browser_development
- browser_production

mobile:
- mobile_development
- mobile_production

desktop:
- desktop_development
- desktop_production
```

or

4)
```
browser:
- browser
- browser_production

mobile:
- mobile
- mobile_production

desktop:
- desktop
- desktop_production
```

### CODE REVIEW

**Comment by enguerran** [comment]
➍

**Comment by kosssi** [comment]
➍

**Comment by goldoraf** [comment]
I prefer the explicit version: option 3! :+1:

**Comment by GoOz** [comment]
③

**Comment by CPatchane** [comment]
Also ⓷, more explicit way 👍

**Comment by y-lohse** [comment]
:three: is explicit and consistent

File: .travis.yml [hunk 1]
```diff
@@ -28,6 +28,6 @@ before_install:
 - echo "password = $TX_PASSWD" >> ~/.transifexrc
 script:
 - yarn test
-- yarn run build
+- yarn build:browser
```
Comment (m4dz) [inline .travis.yml]: Are we sure commands can be run w/o the `run` command, as a shortcut? Yarn doc doesn't mention it…

File: README.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ You can easily view your current running app, you can use the [cozy-stack docker
 ```sh
 # in a terminal, run your app in watch mode
 $ cd cozy-files-v3
-$ yarn run watch
+$ yarn watch:browser
```
Comment (m4dz) [inline README.md]: Same as for `yarn build:browser` command

File: config/webpack.base.config.js [hunk 3]
```diff
@@ -5,8 +5,8 @@ const HtmlWebpackPlugin = require('html-webpack-plugin')
 
 const pkg = require(path.resolve(__dirname, '../package.json'))
 
-const build = process.env.NODE_ENV === 'production'
-const mobile = process.env.MOBILE || false
+const build = /production/.test(process.env.NODE_ENV)
```
Comment (m4dz) [inline config/webpack.base.config.js]: We probably should test `/production$/` to force the test on the suffix part

File: config/webpack.base.config.js [hunk 4]
```diff
@@ -5,8 +5,8 @@ const HtmlWebpackPlugin = require('html-webpack-plugin')
...
+const mobile = /mobile/.test(process.env.NODE_ENV)
```
Comment (m4dz) [inline config/webpack.base.config.js]: In a similar way, we probably want to test `/^mobile/` to test the prefix only

File: .travis.yml (hunk 1 above)
Comment (kosssi) [inline .travis.yml]: yes, it's @goldoraf who learn me. Have you a reference @goldoraf ?

**Comment by kosssi** [comment]
@m4dz I'm ok with your 1. point.

For the 2. point I think it's for another PR ;)

File: .travis.yml (hunk 1 above)
Comment (enguerran) [inline .travis.yml]: I type `yarn <script>` everytime and I did not face yet an error. But it is not documented explicitly in the documentation.

**Comment by kosssi** [comment]
Finally we use this syntax:

```
browser:
- browser:development
- browser:production

mobile:
- mobile:development
- mobile:production

desktop:
- desktop:development
- desktop:production
```

**Comment by m4dz** [comment]
👏👏👏

I'm ok with your point 2. I have a PR incoming w/ cleaning for build and deps, I include it in! Thanks

**Pull request merged**